pkg/loader/loadertest/fake_loader.go

```diff
@@ -30,11 +31,15 @@ type FakeLoader struct {
 }
 
 // NewFakeLoader returns a Loader that delegates calls, and encapsulates
-// a fake file system that the Loader reads from . "initialFilePath" parameter
-// must be an absolute path.
+// a fake file system that the Loader reads from. "initialFilePath" parameter
+// must be an absolute path, and it can be a directory if it has a
+// trailing slash: "/home/seans/project/" (dir) -- "/home/seans/project" (file)
 func NewFakeLoader(initialFilePath string) FakeLoader {
 	// Create fake filesystem and inject it into initial Loader.
 	fakefs := fs.MakeFakeFS()
+	if strings.HasSuffix(initialFilePath, "/") {
+		fakefs.Mkdir(initialFilePath, 0x777)
+	}
```

Comment:
I think fakeFS probably doesn't even care about the dirs. You can store a file at a path for which directory doesn't even exist in fakeFS.?

Reply:
I'm starting to wonder if we should have the fake fs create a `/tmp/something` and use the actual fs. The fake is going to have many limitations that we can't program easily.